projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-checkout.ts

```diff
@@ -128,38 +119,41 @@ export function selectCreditCardPayment() {
 }
 
 export function selectAccountShippingAddress() {
+  const getCartWithDeliveryAddress = interceptCartWithDeliveryAddressEndpoint();
+  const updateAddress = interceptUpdateAddressEndpoint();
+  const getCheckoutDetails = interceptCheckoutB2BDetailsEndpoint();
+  const pudDeliveryMode = interceptPutDeliveryModeEndpoint();
+
   cy.get('.cx-checkout-title').should('contain', 'Delivery Address');
   cy.get('cx-order-summary .cx-summary-partials .cx-summary-row')
     .first()
     .find('.cx-summary-amount')
     .should('not.be.empty');
 
-  cy.intercept(
-    'PUT',
-    `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
-      'BASE_SITE'
-    )}/orgUsers/current/carts/**/addresses/delivery?addressId=*`
-  ).as('updateAddress');
+  //wait for call before updateing address
+  cy.wait(`@${getCartWithDeliveryAddress}`)
+    .its('response.statusCode')
+    .should('eq', 200);
+
+  //wait for updating address
+  cy.wait(`@${updateAddress}`).its('response.statusCode').should('eq', 200);
+
+  //wait for response with updated address
+  cy.wait(`@${getCartWithDeliveryAddress}`)
+    .its('response.statusCode')
+    .should('eq', 200);
 
   cy.get('cx-card').within(() => {
```

Comment:
I would say we don't need to add line 133 to 144, but we can keep.

But let me know of your opinion. Reason why we don't need is we're not waiting for anything data specific and the test is already deterministic. But the real reason is cy.get('cx-card') will wait for 30s anyhow for the card to appear. Although, I agree having extra checks for the response to actually come back is useful, but for this case, I really don't think we need it.

P.S. As mentioned in some other comment, but replace `getCartWithDeliveryAddress` with `getCheckoutDetails` if you decide to keep line 133 to 144.